### ISSUE TITLE
Plan for DeepTrade?

### ISSUE THREAD

**Issue by EmbraceLife** [happynoom/DeepTrade#1]
Hi, Thanks for making it an independent repo! it will make fork and sync with your rep easier. 

I think your intention to contribute DeepTrader is great and I think your deepTrade example is brilliant. As you mentioned that you have reasons to not share the standard version or state of art version of DeepTrade, I am also sympathetic to your reasons. 

As independent repo, you can share your model in whatever version you like. However, if I were the team behind the maintainers of Tensorflow models repo, I would be more likely to accept new models with two conditions met. 

Condition1: state-of-art performance among similar open sourced models
Suppose there are 3 open-sourced repos on DeepTrader, and yours should be the best or state of art performance; if your model is the only open-source repo available, it still should have to be the best you can come up with. 

Condition2: the model initiator (meaning you or your team) is reaching the limit, and would like people around world to help develop this model altogether (this half part is your intention right?).
This means you and your team is already sharing the best you can do (Not the `secretive` or exact parameter or hyper-parameter values of course, but need to share the complex model structure you experimented to achieve state-of-art performance), so that other interested experts can build upon it and improve it further. 

In other words, to be genuine open source, you are willing to share the best you can do and then others can be encouraged to contribute and share the best they can do to this model. 

I do hope Tensorflow models can accept your model sooner, and again I do appreciate your sharing of this simple version of DeepTrader. This model is interesting and I will contribute to it whenever it is possible.

**Comment by happynoom** [comment]
@EmbraceLife Thank you for all your advice. It helps me a lot. The standard version of this project has more layers, including both dropout and more lstm layers. And It is built using keras (easier to implement) with tensorflow backend. To train the deeper model, larger dataset is employed and parameters are carefully picked. We achieve 0.16 percent of profit on average each day on the validation set, better than the simplified model (0.12). 'dropout', 'lstm' and 'batch normalization' are brilliant people's brilliant work. However, I want to demonstrate the 'risk estimation' loss function, which could enable us to optimizing our trade gain directly. Using this loss function, a lot of trading scenes could be carried out by the robot, and the profit(or loss) could be estimated in advance. I wish more researchers observe that, and maybe use it to develop cool trade tools. I am an engineer, busy with jobs, do machine learning experiments in spare, have little time to write a paper on it. As for the standard version of this project, I started converting it to tf, but may not be completed so soon. I should thank you sincerely again. I hope in the future, the tf models accept this work.

**Comment by EmbraceLife** [comment]
Your standard version sounds fantastic, and it is great to know you are working towards open-source your state-of-art model! 

Converting from keras to tensorflow, hopefully gets easier these days to convert from `keras` to `tf.contrib.keras`, given keras has been rewritten from ground up in tensorflow. I spend much time on tensorflow and keras recently, if you could open source your standard version in keras now, I would like to share some of your workload in converting it from keras to tf.contrib.keras. 

Or if you have any other ideas on how I can contribute to your repo, please let know.

**Comment by happynoom** [comment]
@EmbraceLife I am so glad to hear that. I will submit the standard keras project with entire experimental dataset. And also with a best copy of my experimental results. Maybe in this or next weekend with some summarizing work. Thank you very much!

**Comment by happynoom** [comment]
@EmbraceLife Sorry for this long wait. I updated the keras version and added some experimental result. I have submit the keras version of this DeepTrade project. As I said, the validation loss reach -0.16 during train, and I also submit a larger dataset, best model I have got and the train log. The repository is at https://github.com/happynoom/DeepTrade_keras. I appreciate your review and suggestions. Thank you very much!
